src/atomic64/mod.rs

```diff
@@ -15,18 +15,67 @@
 #[cfg(not(feature = "nightly"))]
 mod fallback;
 #[cfg(not(feature = "nightly"))]
-pub use self::fallback::{AtomicF64, AtomicU64};
+pub use self::fallback::{AtomicF64, AtomicI64, AtomicU64};
 
 #[cfg(feature = "nightly")]
 mod nightly;
 #[cfg(feature = "nightly")]
-pub use self::nightly::{AtomicF64, AtomicU64};
+pub use self::nightly::{AtomicF64, AtomicI64, AtomicU64};
+use std::cmp::*;
+use std::ops::*;
 
-pub trait Atomic<T> {
-    fn new(val: T) -> Self;
-    fn set(&self, val: T);
-    fn get(&self) -> T;
-    fn inc_by(&self, delta: T);
+pub trait Number
+    : Sized + AddAssign + SubAssign + PartialOrd + PartialEq + Copy + Send + Sync
+    {
+    /// `std::convert::From<i64> for f64` is not implemented, so that we need to implement our own.
+    fn from_i64(v: i64) -> Self;
+
+    fn into_f64(self) -> f64;
+}
+
+impl Number for i64 {
+    #[inline]
+    fn from_i64(v: i64) -> Self {
+        v
+    }
+
+    #[inline]
+    fn into_f64(self) -> f64 {
+        self as f64
+    }
+}
+
+impl Number for u64 {
+    #[inline]
+    fn from_i64(v: i64) -> Self {
+        v as u64
```

Comment:
What if v is negative?

Reply:
It will not. All from_i64 calls are (0) or (1) in fact. However this is indeed somehow misleading.

- If we name it as "from_u64" -> what if v > max_i64 + 1 && v < max_u64?
- If we make two functions, i.e. from_zero && from_one, it will be too verbose.

 ideas?